### DISCUSSION TITLE
Adding support for http protocol in elasticexporter

### DISCUSSION THREAD

**Discussion by stanosamek** [open-telemetry/opentelemetry-collector-contrib#7965] [Q&A]
Hi, 

I would like to ask if it's planned to add support for http protocol in elasticexporter. Problem is that we have our Elastic APM after F5 which is not supporting grpc protocol.

Thank you

**Comment by jpkrohling** [comment]
cc @axw @simitt @jalvz

**Comment by axw** [comment]
Thanks for the ping!

@stanosamek [elasticexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter) operates over HTTP, but we don't recommend this anymore. Since 7.13 Elastic APM Server has had native support for OTLP/gRPC, which is recommended. We plan to add support for OTLP/HTTP in a future release, and you can subscribe to https://github.com/elastic/apm-server/issues/6128 for updates.

In the mean time, if you can add opentelemetry-collector after F5 then you can configure it with an OTLP/HTTP receiver + OTLP/gRPC exporter to Elastic APM Server.

**Reply by stanosamek** [reply]
@axw I understand that we can configure Otel Collector after F5, but with this setup we are not able to build up high availability solution of Elastic APM. Am I right?

**Reply by axw** [reply]
Right, I see. I'm not that familiar with the options for opentelemetry-collector & load balancing / high availability. You might use https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/loadbalancingexporter, but I'm not sure if that suits your needs. Otherwise you may need a separate gRPC/HTTP2-aware proxy, or just wait for https://github.com/elastic/apm-server/issues/6128. We're hoping to have that implemented in the next few months, but no promises.
